src/generic-container.test.ts

```diff
@@ -33,6 +34,31 @@ describe("GenericContainer", () => {
     await expect(fetch(url)).rejects.toThrowError();
   });
 
+  it("should set network mode", async () => {
+    const container = await new GenericContainer("cristianrgreco/testcontainer", "1.1.11")
+      .withName("special-test-container")
+      .withNetworkMode("host")
+      .start();
+
+    const dockerodeClient = new Dockerode()
+    const listedContainers = await dockerodeClient.listContainers()
+
+    let containerInfo: undefined | Dockerode.ContainerInfo
+    for (const listedContainer of listedContainers) {
+      if (listedContainer.Names.includes("/special-test-container")) {
+        containerInfo = listedContainer
```

Comment:
The way we obtain `containerInfo` can be drastically simplified if something like what the PR #46 provides is implemented.

Reply:
I'd like to keep the tests as 'black box' tests, where we test a behaviour. It is for this reason as well that we don't currently use Dockerode in our tests. This test only checks that a configuration parameter is passed to Dockerode.

Reply:
It does not check that a configuration is passed to Dockerode, it tests that the configuration reached the docker daemon.

Dockerode obtains this information on its own making a call to the docker daemon, not recovering the objects that we created previously.

Reply:
My point is that the existing tests test a given behaviour. If we swapped dockerode for something else the tests would still pass. Ideally we shouldn't have to dig into the internals to test that something works.

Reply:
I'm not digging into the internals, I use a tool that, luckily for me, you also use for the internals. But I'm not accessing the internals of the classes in any way.